$NetBSD: patch-ab,v 1.4 2005/04/13 18:12:55 rillig Exp $

The C++ standard library headers shipped with gcc-2.95 sometimes include
<_G_config.h>, which undefines the NULL macro and then includes
<stddef.h> afterwards. Currently the situation is:

  #include <stddef.h>		/* defines NULL */
  #include <_G_config.h>	/* undefines NULL */
  #  include <stddef.h>		/* is skipped (guarded) */
  => NULL is undefined.

So we need to include <_G_config.h> at the very beginning. Then the
sequence is:

  #include <_G_config.h>	/* undefines NULL */
  #  include <stddef.h>		/* defines NULL */
  #include <_G_config.h>	/* is skipped (guarded) */
  => NULL is defined.

--- src/sys.h.in.orig	Sun Jun 13 16:50:10 2004
+++ src/sys.h.in	Tue Apr 12 11:53:19 2005
@@ -40,6 +40,11 @@
 #define _THREAD_SAFE
 #endif
 
+/* This is needed to have a NULL macro */
+#if defined(__NetBSD__) && defined(__GNUC__) && ((__GNUC__) < 3)
+# include <stl_config.h>
+#endif
+
 #ifndef __linux__
 #include <sys/types.h>
 #endif
